libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -250,6 +250,95 @@ string ABIFunctions::cleanupFunction(Type const& _type, bool _revertOnFailure)
 			break;
 		}
 		case Type::Category::Enum:
+		{
+			size_t members = dynamic_cast<EnumType const&>(_type).numberOfMembers();
+			solAssert(members > 0, "empty enum should have caused a parser error.");
+			templ("body", "cleaned := and(value, " + toCompactHexWithPrefix(members) + ")");
```

Comment:
I don't think there is a reasonable way to clean an enum. Out of range is out of range and it should revert.

Also, I think in it's current form, it is missing a `1<<`.